docs/pattern-matching.md

```diff
@@ -60,20 +60,20 @@ They offer more advanced pattern matching capabilities and also take care linear
 Pattern matching equations are transformed into a tree of `match` and `switch` terms from left to right.
 ```py
 # These two are equivalent
-(Foo 0 false (Cons h1 (Cons h2 t))) = (A h1 h2 t)
-(Foo 0 * *) = B
+(Foo 0 false (Cons h1 (Cons h2 t))) = (Bar h1 h2 t)
```

Comment:
The syntax for constructors in this file is wrong everywhere, should be Bool/False, List/Cons, Maybe/Some, etc